languages/python/oso/polar/host.py

```diff
@@ -20,25 +20,56 @@
 class Host:
     """Maintain mappings and caches for Python classes & instances."""
 
-    def __init__(self, polar, classes=None, instances=None, get_field=None):
+    def __init__(
+        self,
+        polar,
+        classes=None,
+        cls_names=None,
+        instances=None,
+        get_field=None,
+        types=None,
+        fetchers=None,
+    ):
         assert polar, "no Polar handle"
         self.ffi_polar = polar  # a "weak" handle, which we do not free
         self.classes = (classes or {}).copy()
+        self.cls_names = (cls_names or {}).copy()
         self.instances = (instances or {}).copy()
+        self.types = (types or {}).copy()
+        self.fetchers = (fetchers or {}).copy()
         self._accept_expression = False  # default, see set_accept_expression
 
-        def default_get_field(_obj, _field):
-            raise PolarRuntimeError("Cannot generically walk fields of a Python class")
+        # Check the types.
+        def default_get_field(obj, field):
+            return self.types_get_field(obj, field)
 
         self.get_field = get_field or default_get_field
 
+    # @Q: I'm not really sure what I'm returning here.
+    def types_get_field(self, obj, field):
+        obj_type_name = self.cls_names[obj]
+        if obj_type_name in self.types:
+            obj_type_info = self.types[obj_type_name]
+            if field in obj_type_info:
```

Comment:
It checks the registered type info to see what the type of the field lookup is. I'll add a comment.

Reply:
This also will probably go away later. Eventually the type info will be passed into the core and we wont need this anymore it can all be resolved there (not for this release though).